### DISCUSSION TITLE
this project support centos7 build for it?

### DISCUSSION THREAD

**Discussion by streamFunc** [paullouisageneau/libdatachannel#790] [Q&A] [answered]
hi all,
i would like to build this project use centos7 gcc 4..8.5,but report this question:
addr.c:103:7: error: ‘const struct in6_addr’ has no member named ‘s6_addr32’
addr.c:196:35: error: ‘NI_NUMERICSERV’ undeclared (first use in this function)
and so on lots of build question...
and i use ubuntu20.04 build for it is ok.
would this project support centos7 build for it?

**Comment by Dw9** [comment, marked as answer]
support.
You need to upgrade your gcc version to 7 or above,cause this project is based on c++17.

you can follow this to upgrade:
sudo yum install centos-release-scl
sudo yum install devtoolset-8-gcc*
scl enable devtoolset-8 bash
gcc -v
<img width="507" alt="image" src="https://user-images.githubusercontent.com/36155473/215257602-6ecb4235-78d6-4188-9abf-3bc412a27bd5.png">

**Reply by streamFunc** [reply]
thank you.i follow it to upgrade gcc and solve this question.but another question
certificate.cpp:274:23: error: ‘X509_getm_notBefore’ was not declared in this scope
certificate.cpp:275:23: error: ‘X509_getm_notAfter’ was not declared in this scope


how can i solve this？upgrade  openssl version?

**Reply by Dw9** [reply]
yes。The default openssl version on centos7 is relatively low.

Step 1:upgrade openssl version
$ wget https://www.openssl.org/source/openssl-1.1.1k.tar.gz
$ tar -xvzf openssl-1.1.1k.tar.gz
$ yum install -y zlib zlib-devel perl-CPAN
$ ./config shared --openssldir=/usr/local/openssl --prefix=/usr/local/openssl
$ make && make install
$ echo "/usr/local/lib64/" >> /etc/ld.so.conf
$ echo "/usr/local/openssl/lib" >> /etc/ld.so.conf
$ ldconfig
$ ln -s /usr/local/openssl/bin/openssl  /usr/local/bin/openssl

Step 2:Specify the path of openssl
$cmake ..   -DOPENSSL_ROOT_DIR=/usr/local/openssl  -DOPENSSL_LIBRARIES=/usr/local/openssl/lib

**Reply by streamFunc** [reply]
thank you,it work

**Reply by streamFunc** [reply]
hi,after build for it success ,i can not find datachannel-static lib,how can i set CMakeLists for datachannel-static?
